src/base/static/components/form-fields/rich-textarea-field.js

```diff
@@ -0,0 +1,203 @@
+import React, { Component } from "react";
+import cx from "bem-classnames";
+import ReactQuill, { Quill } from "react-quill";
+const BlockEmbed = Quill.import("blots/block/embed");
+const SnowTheme = Quill.import("themes/snow");
+const Link = Quill.import("formats/link");
+
+const Util = require("../../js/utils.js");
+
+const baseClass = "mapseed-rich-textarea-field";
+
+// NOTE: this routine is taken from Quill's themes/base module.
+const extractVideoUrl = (url) => {
+  let match = url.match(/^(?:(https?):\/\/)?(?:(?:www|m)\.)?youtube\.com\/watch.*v=([a-zA-Z0-9_-]+)/) ||
+              url.match(/^(?:(https?):\/\/)?(?:(?:www|m)\.)?youtu\.be\/([a-zA-Z0-9_-]+)/);
+  if (match) {
+    return (match[1] || 'https') + '://www.youtube.com/embed/' + match[2] + '?showinfo=0';
+  }
+  if (match = url.match(/^(?:(https?):\/\/)?(?:www\.)?vimeo\.com\/(\d+)/)) {
+    return (match[1] || 'https') + '://player.vimeo.com/video/' + match[2] + '/';
+  }
+  return url;
+};
+
+class WrappedVideo extends BlockEmbed {
```

Comment:
Perhaps we can avoid inheriting BlockEmbed, and composing it instead?

See here for further reading:
https://reactjs.org/docs/composition-vs-inheritance.html

>At Facebook, we use React in thousands of components, and we haven’t found any use cases where we would recommend creating component inheritance hierarchies.

Reply:
`BlockEmbed` isn't a React component, though-- it's part of the core Quill framework. Not sure if that makes a difference here?

Reply:
Ah, I thought it was a React component. I'm not familiar with the Quill api, but you can disregard my earlier comment.